### PR TITLE
MAINT: fix linter issues

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
#### Reference issue

https://github.com/scipy/scipy/pull/19295#issuecomment-1732692133

#### What does this implement/fix?

These errors prevent from (eventually) merging #19295.

### CODE REVIEW

File: scipy/optimize/tests/test__shgo.py [hunk 1]
```diff
@@ -27,7 +27,7 @@ def __init__(self, bounds, expected_x, expected_fun=None,
 def wrap_constraints(g):
     cons = []
     if g is not None:
-        if (type(g) is not tuple) and (type(g) is not list):
+        if not isinstance(g, tuple | list):
```
Comment (j-bowhay) [inline scipy/optimize/tests/test__shgo.py:30]:
```suggestion
        if not isinstance(g, (tuple, list)):
```

File: scipy/optimize/_shgo_lib/_complex.py [hunk 2]
```diff
@@ -137,8 +137,7 @@ def __init__(self, dim, domain=None, sfield=None, sfield_args=(),
             self.min_cons = constraints
             self.g_cons = []
             self.g_args = []
-            if (type(constraints) is not tuple) and (type(constraints)
-                                                     is not list):
+            if not isinstance(constraints, tuple | list):
```
Comment (j-bowhay) [inline scipy/optimize/_shgo_lib/_complex.py:140]:
```suggestion
            if not isinstance(constraints, (tuple, list)):
```

**Review by j-bowhay — CHANGES_REQUESTED** [review]

File: scipy/optimize/tests/test__shgo.py (hunk 1 above)
Comment (DimitriPapadopoulos) [reply scipy/optimize/tests/test__shgo.py]: Indeed, [union types should work as well](https://docs.python.org/3/library/functions.html#isinstance), but only in **Python ≥ 3.10**:
> If _classinfo is_ a tuple of type objects (or recursively, other such tuples) or a [Union Type](https://docs.python.org/3/library/stdtypes.html#types-union) of multiple types, return `True` if _object_ is an instance of any of the types.

Comment (j-bowhay) [reply scipy/optimize/tests/test__shgo.py]: See: https://numpy.org/neps/nep-0029-deprecation_policy.html for python version roadmap

**Review by j-bowhay — APPROVED** [review]
Thanks @DimitriPapadopoulos

**Pull request merged**